core/types/rollup_cost.go

```diff
@@ -0,0 +1,285 @@
+// Copyright 2022 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+package types
+
+import (
+	"bytes"
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/log"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+const (
+	// The two 4-byte Ecotone fee scalar values are packed into the same storage slot as the 8-byte
+	// sequence number and have the following Solidity offsets within the slot. Note that Solidity
+	// offsets correspond to the last byte of the value in the slot, counting backwards from the
+	// end of the slot. For example, The 8-byte sequence number has offset 0, and is therefore
+	// stored as big-endian format in bytes [24:32] of the slot.
+	BaseFeeScalarSlotOffset     = 12 // bytes [16:20] of the slot
+	BlobBaseFeeScalarSlotOffset = 8  // bytes [20:24] of the slot
+
+	// scalarSectionStart is the beginning of the scalar values segment in the slot
+	// array. baseFeeScalar is in the first four bytes of the segment, blobBaseFeeScalar the next
+	// four.
+	scalarSectionStart = 32 - BaseFeeScalarSlotOffset - 4
+)
+
+func init() {
+	if BlobBaseFeeScalarSlotOffset != BaseFeeScalarSlotOffset-4 {
+		panic("this code assumes the scalars are at adjacent positions in the scalars slot")
+	}
+}
+
+var (
+	// BedrockL1AttributesSelector is the function selector indicating Bedrock style L1 gas
+	// attributes.
+	BedrockL1AttributesSelector = []byte{0x01, 0x5d, 0x8e, 0xb9}
+	// EcotoneL1AttributesSelector is the selector indicating Ecotone style L1 gas attributes.
+	EcotoneL1AttributesSelector = []byte{0x44, 0x0a, 0x5e, 0x20}
+
+	// L1BlockAddr is the address of the L1Block contract which stores the L1 gas attributes.
+	L1BlockAddr = common.HexToAddress("0x4200000000000000000000000000000000000015")
```

Comment:
It looks like the L1Block address of Kroma should be in there.
```suggestion
	L1BlockAddr = common.HexToAddress("0x4200000000000000000000000000000000000002")
```

Reply:
Nice catch!